contracts/AirbroFactorySMCampaign.sol

```diff
@@ -0,0 +1,132 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.14;
+
+import "./airdropsSMCampaign/TokenDropSMCampaign.sol";
+import "./airdropsSMCampaign/ExistingTokenDropSMCampaign.sol";
+import "./airdropsSMCampaign/Existing1155NftDropSMCampaign.sol";
+import "./Airbro1155Contract.sol";
+
+
+/// @title Airbro - NFT airdrop tool factory contract - for owners of 1155 Nfts
+contract AirbroFactorySMCampaign {
+    // index of deployed airdrop contracts
+    address[] public airdrops;
+    uint256 public totalAirdropsCount = 0;
+    address public admin = 0xF4b5bFB92dD4E6D529476bCab28A65bb6B32EFb3;
+
+    address[] public nft1155Contracts;
+    uint256 public totalNft1155ContractsCount = 0;
+
+    event NewAirdrop(address indexed rewardedNftCollection, address indexed airdropContract, address indexed airdropCreator);
+    event NewNft1155Contract(address indexed nft1155Contract, address indexed contractCreator);
+    
+    event AdminChanged(address indexed adminAddress);
+
+    error NotAdmin();
+
+    modifier onlyAdmin(){
+        if(msg.sender != admin) revert NotAdmin();
+        _;
+    }
+
+    constructor() {}
+
+    /// @notice Creates a new ERC1155 collection
+    /// @param uri - IPFS link to the NFT image
+    function createNewNft1155Contract (
+        string memory uri
+    ) public {
+        Airbro1155Contract nft1155Contract = new Airbro1155Contract(
+            uri
+        );
+
+        nft1155Contracts.push(address(nft1155Contract));
+        totalNft1155ContractsCount++;
```

Comment:
For stuff like this (when there is no risk of overflow) you can use unchecked to save some gas
        unchecked {
            i++;
        }